sid/simulate.py

```diff
@@ -302,20 +340,38 @@ def _process_initial_states(states, assort_bys):
     for col in COUNTDOWNS:
         if col not in states.columns:
             states[col] = -1
-        states[col] = states[col].astype(DTYPE_COUNTER)
+        states[col] = states[col].astype(DTYPE_COUNTDOWNS)
 
-    states["infection_counter"] = 0
+    states["infection_counter"] = DTYPE_INFECTION_COUNTER(0)
 
     for model_name, assort_by in assort_bys.items():
         states[f"group_codes_{model_name}"], _ = factorize_assortative_variables(
             states, assort_by
         )
 
-    return states, index_names
+    return states
+
 
+def _dump_periodic_states(states, output_directory, date):
+    group_codes = states.filter(like="group_codes_").columns.tolist()
+    useless_columns = USELESS_COLUMNS + group_codes
+
+    states.copy(deep=True).drop(columns=useless_columns).to_parquet(
```

Comment:
I don't think you need to make a copy here. `df.drop` returns a copy and after this we jump to the next period and overwrite many columns anyways. Or is there something I am overlooking?